packages/fuel-gauge/src/transaction.test.ts

```diff
@@ -21,4 +23,120 @@ describe('Transaction', () => {
 
     expect(tx.type).toBe(TransactionType.Mint);
   });
+
+  it('should spent message with data just fine (W/ AMOUNT)', async () => {
+    const messageAmount = 100_000;
+    const initialBalance = 50_000;
+
+    const testMessage = new TestMessage({
+      data: hexlify(InputMessageCoder.encodeData('0x09')),
+      amount: messageAmount,
+    });
+
+    using launched = await launchTestNode({
+      nodeOptions: {
+        args: ['--poa-instant', 'false', '--poa-interval-period', '1ms'],
+      },
+      contractsConfigs: [
+        {
+          factory: CallTestContractFactory,
+        },
+      ],
+      walletsConfig: {
+        count: 1,
+        amountPerCoin: initialBalance,
+        messages: [testMessage],
+      },
+    });
+
+    const {
+      provider,
+      contracts: [contract],
+      wallets: [fundedWallet],
+    } = launched;
+
+    const baseAssetId = provider.getBaseAssetId();
+
+    const {
+      messages: [message],
+    } = await fundedWallet.getMessages();
+
+    const request = await contract.functions.foo(10).getTransactionRequest();
+    request.addMessageInput(message);
```

Comment:
Why do we need to add the message input here if the wallet is already funded with the [`testMessage`](https://github.com/FuelLabs/fuels-ts/pull/3078/files#diff-3ccf1abbc4dc63f3169e41bf35f6a65f268903ebf6185701aff6ec308e975ce7R31) ?

Reply:
@maschad But it is not funded.

Also, messages with data are not considered resources since they cannot pay for the TX fee.

Because of this the `getResourcesToSpend` will not return any message with data.

Reply:
I realize now that a consumer has to explicitly state that the request should be funded with the message input. Thanks for clarifying.